src/extension.ts

```diff
@@ -255,6 +257,10 @@ export async function activate(context: vscode.ExtensionContext) {
 
         recordToolkitInitialization(activationStartedOn, getLogger())
 
+        fs.watchFile(getCredentialsFilename(), async () => {
```

Comment:
There are a few concerns with this:
1. Credentials can be configured in the config file as well
2. Does this silently try to login? What happens if it fails? This could get very spammy if it keeps trying to login with bad credentials.
3. How does `watchFile` interact across various operating systems? I know it polls the file on a set interval (default is ~5s), but when does it fire? Documentation says `The callback listener will be called each time the file is accessed.` Does that mean anytime I open the file it will try to login?

Reply:
1.  Watcher now watches the folder, but we only act if the change is on `credentials` or `config`
2. Yes, this does silently try to login, but only makes one attempt. 
3. Good question.  I looked deeper into this and the options.  Switched to `fs.watch` instead of `fs.watchFile` for a number of reasons. The first and most important is that `fs.watchFile` calls `stat` at every interval (yes, the default is 5s), which means it's consuming the CPU even if no changes are made. `fs.watch` listens for an event and passes the event type and filename, which means we only act if there is an actual change to those files.  It works on all major OSes and the only one listed not to work is IBM i systems. I'll add a try-catch around this so that in the event of an error, it will default to the old behavior.